qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -142,7 +142,5 @@ def _compose_layouts(self, initial_layout, pass_final_layout, qregs):
         """
         trivial_layout = Layout.generate_trivial_layout(*qregs)
         qubit_map = Layout.combine_into_edge_map(initial_layout, trivial_layout)
-        final_layout = {
-            v: pass_final_layout[qubit_map[v]] for v, _ in initial_layout.get_virtual_bits().items()
-        }
+        final_layout = {v: pass_final_layout[qubit_map[v]] for v in initial_layout._v2p}
```

Comment:
It looks like there's still a `Layout.__getitem__` here in `pass_final_layout`. Would it make sense to avoid that? Also, can the `initial_layout._v2p` fetch be moved outside the loop?

Reply:
Oh that's a good call I missed that here. Yeah we should avoid that. As for the `initial_layout._v2p` fetch we actually can't move it outside of the loop because each iteration updates the `initial_layout` from the routing pass's final layout. So it changes on each iteration

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/95caec850a9ee4bf964a8f6fb1be3aafe774f971